### PR TITLE
scripts: add vkey upload script

### PR DESCRIPTION
**Pull request by akirillo**
This PR introduces a new script for uploading verification keys, using the circuit definitions from the relayer. Ad-hoc testing of the script shows that it works (i.e., the transaction is executed), though ad-hoc integration tests with the relayer's Arbitrum client don't have on-chain verification succeeding. Fixing that is out of scope for this PR.

### CODE REVIEW

File: scripts/src/cli.rs [hunk 1]
```diff
@@ -125,3 +130,28 @@ pub struct UpgradeArgs {
...
+    #[arg(short, long)]
+    pub circuit: Circuit,
+
+    /// The address of the darkpool proxy contract
+    #[arg(short, long)]
+    pub darkpool_address: String,
+
+    /// Whether or not to use the testing circuit size parameters
+    #[arg(short, long)]
+    pub test: bool,
```
Comment (joeykraut) [inline scripts/src/cli.rs]: maybe `small` instead, test implies that the command is being run as a test, small suggests the actual effect of this flag

File: scripts/src/utils.rs [hunk 2]
```diff
@@ -175,3 +191,73 @@ pub fn deploy_stylus_contract(
...
+            .map_err(|_| ScriptError::ConversionError)?,
+        q_comms: try_unwrap_commitments(&jf_vkey.selector_comms)?,
+        sigma_comms: try_unwrap_commitments(&jf_vkey.sigma_comms)?,
+        g: jf_vkey.open_key.g,
+        h: jf_vkey.open_key.h,
+        x_h: jf_vkey.open_key.beta_h,
+    })
+}
+
+pub fn get_vkey_bytes(circuit: Circuit, test: bool) -> Result<Vec<u8>, ScriptError> {
```
Comment (joeykraut) [inline scripts/src/utils.rs]: Perhaps `vkey_to_bytes`, `get` implies this is some field of the verifier key

File: scripts/src/utils.rs [hunk 3]
```diff
@@ -175,3 +191,73 @@ pub fn deploy_stylus_contract(
...
+) -> Result<[G1Affine; N], ScriptError> {
+    comms
+        .iter()
+        .map(|c| c.0)
+        .collect::<Vec<_>>()
+        .try_into()
+        .map_err(|_| ScriptError::ConversionError)
+}
+
+pub fn convert_jf_vkey(jf_vkey: VerifyingKey<Bn254>) -> Result<VerificationKey, ScriptError> {
```
Comment (joeykraut) [inline scripts/src/utils.rs:209]: This should have a doc comment or be renamed. Consider the perspective of someone who is reading about our system for the first time, what is `jf`, what conversion is going on here, is it runtime type? These things are obvious to us but won't be to a first time reader

**Review by joeykraut — APPROVED** [review]

File: scripts/src/utils.rs (hunk 2 above)
Comment (akirillo) [reply scripts/src/utils.rs]: Ack, I'll do `gen_vkey_bytes`, don't want the method name to imply that a vkey should be passed in

**Pull request merged**